research-software-directory/RSD-as-a-service
# 359 Custom theme, footer logo, contact email and links

# Custom home page for Helmholtz and theme provider using settings.json.

Custom settings can be uploaded using volume mounts, see README in deployment folder.

Closes #359 
Closes #374 
Closes #410 
Closes #438

## Changes proposed in this pull request
* The theme approach is adapted to load different, predefined, themes at the runtime using volume mounts of frontend image.
* The homepage is loaded based on the host value provided in the mounted settings.json file. If host is not provided in the settings.json the default rsd homepage is loaded.
* If `settings.json` is not provided via volume mount the default settings file is loaded (public/data/settings.json)
* if `index.css` file with font imports is not provided via volume mount the default index.css file is used (public/styles/index.css)
* When using default RSD settings one does not need to provide volume mounts in the frontend service
* Footer links can be added using settings.json
* Footer logo and contact email are defined in settings.json

How to test:
* delete the directories `frontend/{node_modules,.next}` if they exist because they will be recreated by the frontend docker development container (and with the ownership of the docker system user, which most likely will be `root`, so changing the directory ownership with `chown` should also be possible instead of deleting them)
* `docker-compose build` to build the solution
* create `hmz` folder in the deployment directory. Note I have added hmz folder in the .gitignore file
* Copy all folder and files from the rsd folder into hmz folder. Provide your values in the index.css, settings.json files and upload your custom images used on your homepage. * 
* change values in docker-compose frontend service to:

```yaml
  # frontend definitions example to mount custom theme
  # add volumes as in this example to load index.css and settings.json
  frontend:
    ...other definitions
    volumes:
      # mount local styles folder into styles folder of frontend app (index.css file is the access point)
      - ./deployment/hmz/styles:/app/public/styles
      # mount custom settings (theme colors and typography definitions) into data folder of frontend app
      - ./deployment/hmz/data:/app/public/data
      # mount custom images used on custom home page (for helmholtz only)
      - ./deployment/hmz/images:/app/public/images
    ...other definitions
```
PR Checklist:
